### PR TITLE
[WIP] mgr/dashboard: Add E2E tests which verify that creating/deleting an RGW user works

### PR DESCRIPTION
**Pull request by adk3798**
Signed-off-by: Adam King <adking@redhat.com>
Signed-off-by: Rafael Quintero <rquinter@redhat.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/userCreate.po.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  getAllUsers() {
+    return element.all(by.css('div.datatable-body-cell-label'));
+  }
+
+  getUser() {
+    return element.all(by.css('div.datatable-body-cell-label')).filter(function(elem, index) {
+      return elem.getText().then(function(text) {
+        const i = index;
+        index = i;
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/userCreate.po.ts]: What is the exact purpose of this?

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/userCreate.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    it('should select the new user and delete it', () => {
+      // Checks if the buttons required to delete the users are present and if
+      // the pop-up menu is present, then clicks the appropriate buttons to delete
+      // the selected user. Also checks if the user has been deleted by comparing the
+      // number of users before and after the delete as well as checking that
+      // 'testuser' is no longer present after the delete
+      page.navigateTo();
+      const EC = protractor.ExpectedConditions;
+      let numDigits = 1;
+      const allu = page.getAllUsers();
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/userCreate.e2e-spec.ts]: As we use camelCase, I think "allUsers" var name is more intentional/meaningful.

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/userCreate.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+      let count1 = '0';
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/userCreate.e2e-spec.ts]: The var name "count1" is not very intentional/meaningful, would it be better:
usersBeforeDeletion ?

File: src/pybind/mgr/dashboard/frontend/e2e/rgw/userCreate.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+      const toggle = page.getToggleMenu();
+      const dtitle = page.getBoxTitle();
+      const confirm = page.getConfirm();
+      const delbutton = page.getDelButton();
+      let count2 = '0';
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/e2e/rgw/userCreate.e2e-spec.ts]: The var name "count2" is not very intentional/meaningful, would it be better:
usersAfterDeletion ?

**Comment by adk3798** [comment]
Closing this because it is going to be remade to follow the format laid out in #28592 . The changes made will be significant enough it makes more sense to have a new pull request than edit this one

**Pull request closed without merging**